### ISSUE TITLE
Very few math functions available

### ISSUE THREAD

**Issue by jakubpawlowski** [bazaarvoice/jolt#676]
Why is there 'doubleSum' function but no 'doubleMultiply'. It's not difficult to implement. Here's an example:
```
public static Optional<Double> doubleMultiply(List<Object> args) {
        return Optional.of(args.stream()
                .map(Objects::toDouble)
                .filter(Optional::isPresent)
                .map(Optional::get)
                .reduce(1.0, (a, b) -> a * b));
    }
```
Is there any specific reason for not doing it or should I just raise PR to add a couple of functions.
Similar question was asked in the last comment here:
https://github.com/bazaarvoice/jolt/issues/311

Another very useful one which I can think of would be 'greaterThan' returning a boolean 'true' if the first value is greater than second.

(I followed here an approach taken to functions such as intSubtract())
```
public static Optional<Boolean> greaterThan(List<Object> argList) {

        if ( argList == null || argList.size() != 2 ) {
            return Optional.empty();
        }

        if ( ! ( argList.get(0) instanceof Number && argList.get(1) instanceof Number ) ) {
            return Optional.empty();
        }

        Number a = (Number) argList.get(0);
        Number b = (Number) argList.get(1);

        return Optional.of( new BigDecimal(a.toString()).compareTo(new BigDecimal(b.toString())) > 0 );
    }
```
(or you can write more optimized functions greaterThanDouble, greaterThanInt, etc.)

**Comment by milosimpson** [comment]
The math functions that exist are just what we have needed in BV.    We really needed, them so we built modify, but I really don't like how modify works.   Which is why it's internals are all deprecated.

I have figured out the right approach, but haven't had the time to implement it.

Basically, the idea would be to use an annotations based way of decorating a function, and then Jolt could do the type checking and hand the function what it expects.

As it is now, all the handling of type checking are hardcoded / a mess.
